File: Edison.Web/Edison.AdminPortal/ClientApp/src/app/reducers/response/response.model.ts
import { GeoLocation } from '../../core/models/geoLocation';
import { ActionPlan } from '../action-plan/action-plan.model';
import { Event } from '../event/event.model';

export enum ResponseState {
    Inactive,
    Active
}

export class Response {
    responseId: string;
    responderUserId: string;
    responseState: ResponseState;
    startDate: string;
    endDate: string;
    actionPlan: ActionPlan;
    actionPlanId: string;
    primaryEventClusterId?: string;
    geolocation: GeoLocation;
    eventClusterIds: string[];
    event?: Event;
    name: string;
    icon: string;
    color: string;
    delayStart: boolean = false;

    constructor (event: Event, actionPlan: ActionPlan, user, responseId?: string) {
        this.responderUserId = user.profile[ 'oid' ];
        this.actionPlanId = actionPlan.actionPlanId;
        this.responseState = ResponseState.Active;
        this.eventClusterIds = [];
        this.name = actionPlan.name;
        this.icon = actionPlan.icon;
        this.color = actionPlan.color;
        this.actionPlan = actionPlan;

        if (event) {
            this.primaryEventClusterId = event.eventClusterId;
            this.geolocation = event.device.geolocation;
            this.event = event;
        } else {
            this.delayStart = true;
            this.primaryEventClusterId = null;
        }

        if (responseId) {
            this.responseId = responseId;
        }
    }
}

File: Edison.Core/Edison.Core.Common/Models/Response/ResponseModel.cs
using System;
using System.Collections.Generic;

namespace Edison.Core.Common.Models
{
    public class ResponseModel
    {
        public Guid ResponseId { get; set; }
        public Guid ResponderUserId { get; set; }
        public int ResponseState { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string Name { get; set; }
        public string Color { get; set; }
        public string Icon { get; set; }
        public List<string> SafeUsers { get; set; }
        public ResponseActionPlanModel ActionPlan { get; set; }
        public Guid? PrimaryEventClusterId { get; set; }
        public Geolocation Geolocation { get; set; }
        public IEnumerable<Guid> EventClusterIds { get; set; }
    }
}
